service/history/timerQueueTaskExecutorBase.go

```diff
@@ -260,19 +261,30 @@ func (t *timerQueueTaskExecutorBase) deleteWorkflowHistory(
 
 func (t *timerQueueTaskExecutorBase) deleteWorkflowVisibility(
 	task *persistencespb.TimerTaskInfo,
+	msBuilder mutableState,
```

Comment:
this ms is not needed here

Reply:
but task still needs to be added to ms. no?

Reply:
i may need sometime to think about the `deleteWorkflow`
the existing logic flow does not support what you are trying to do